File: 1. Learn the Basics/1. Pattern/Java/12.java
/* package codechef; // don't place package name! */

/*

1        1
12      21
123    321
1234  4321
1234554321

*/
import java.util.*;
import java.lang.*;
import java.io.*;


/* Name of the class has to be "Main" only if the class is public. */
class Codechef
{
    public void numberCrown(int n) {
        // Write your code here.
        for(int i=1;i<=n;i++){
            for(int j=1;j<=i;j++){
                System.out.print(j);
            }
            for(int j=1;j<=2*(n-i);j++){
                System.out.print(" ");
            }
            for(int j=i;j>0;j--){
                System.out.print(j);
            }
            System.out.println();
        }
    }
	public static void main (String[] args) throws java.lang.Exception
	{
		// your code goes here
		Codechef obj = new Codechef();
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		obj.numberCrown(n);
	}
}

File: 1. Learn the Basics/1. Pattern/Python/12.py
# 1        1
# 12      21
# 123    321
# 1234  4321
# 1234554321

def numberCrown(n):
    for i in range(1,n+1):
        for j in range(1,i+1):
            print(j,end="")
        for j in range(1,2*(n-i)+1):
            print(" ",end="")
        for j in range(i,0,-1):
            print(j,end="")
        print("\r")

        
n = int(input())
numberCrown(n)